### ISSUE TITLE
[Tag list] Stacked, Stacked Over 100 and Stacked Removable tag list have first tag without role

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3007]
### Provide a general summary of the issue here

For [Tag list Stacked](https://storybook.sanomalearning.design/?path=/story/feedback-status-tag-tag-list--stacked), [Tag list Stacked Over 100](https://storybook.sanomalearning.design/?path=/story/feedback-status-tag-tag-list--stacked-over-100) and [Tag list Stacked Removable](https://storybook.sanomalearning.design/?path=/story/feedback-status-tag-tag-list--stacked-removable) first list item (with tooltip) is added inside `<sl-tag-list>` but does not have any `role` specified.
That violates `axe-core` rule "ARIA required children" and makes it harder to understand list relation.

<img width="976" height="43" alt="Image" src="https://github.com/user-attachments/assets/185def66-9675-4961-a011-8dd1f82fa4ee" />

<img width="1018" height="45" alt="Image" src="https://github.com/user-attachments/assets/a45e105a-5fe7-409f-8812-9abaabee65a1" />

<img width="953" height="50" alt="Image" src="https://github.com/user-attachments/assets/bdffea87-c653-4ed0-a339-c959d481ad5c" />

### 🤔 Expected Behavior?

All `<sl-tag>` inside `<sl-tag-list role='list'>` should have `role='listitem'`
The number of list item announced by screenreader should equal to all `<sl-tag>` inside `<sl-tag-list>`

### 😯 Current Behavior

First `<sl-tag>` does not have `role='listitem'`.
The number of list item announced by screenreader equals to "ALL `<sl-tag>` NUMBER - 1" as the first `<sl-tag>` is not viewed as `list item`.
<img width="472" height="304" alt="Screenshot of code with an issue" src="https://github.com/user-attachments/assets/87f917f1-696c-4317-9afa-8ccc5341403f" />

### 💁 Possible Solution

Adding `role='listitem'` to the first `<sl-tag>` inside `<sl-tag-list>`.

### 🖥️ Steps to Reproduce

1. In code find `<sl-tag-list>`
2. Check if the first `<sl-tag>` inside `<sl-tag-list>` has `role='listitem'`

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### What operating system are you using?

macOS and Win11

### 👤 Your name

SLDS team

### 🧢 Your product/team

SLDS team

### 🎨 Your Theme(s)

Sanoma Learning
